### DISCUSSION TITLE
Reference collections for Zigbee Stack and related dev docks.

### DISCUSSION THREAD

**Discussion by MattWestb** [zigpy/zigpy#595] [ZHA development]
## EZSP Zigbee EmberZNet reference release notes from Silabs.
Silabs have putting together some of its release notes for there Zigbee PRO for EM35x EZSP from 4.3.0 GA and newer (perhaps longer back is looking more deeper in the collection) for EM35x and EFR32. 
The KB is [Reference Collection of Ember ZNet and AFV2 Release Notes](https://www.silabs.com/community/wireless/zigbee-and-thread/knowledge-base.entry.html/2017/12/27/reference_collection-PD7Q) and one zip file is attached if the dropbox is going offline. [Ember ZNet and AFV2 Release Notes.zip](https://github.com/zigpy/zigpy/files/5724197/Ember.ZNet.and.AFV2.Release.Notes.zip) is uptodate with EZSP 6.8.X.X. 
And the latest (not in the zip) is here [6.9.0.0 GA](https://www.silabs.com/documents/public/release-notes/emberznet-release-notes-6.9.0.0.pdf).
Its good for devs finding then things in the API was changed and then bugs / workaround / fixes.

One more interesting collection from Silabs that is new made [Zigbee & Thread Knowledge Base Articles List](https://silabs-prod.adobecqms.net/community/wireless/zigbee-and-thread/knowledge-base.entry.html/2020/04/01/zigbee_thread_knowledgebasearticleslist-ih5r) that is pointing out some good reference docks from Silabs.

If finding some more good documents that can helping the devs, pleas pointing it out (if legal possible) so the devs can using the information to making our system better.

The same for all other platforms then more good info its easier for the devs making things working so all info is wanted for all our platforms !!

Sub postings for EZSP madjor version:  [EZSP 6.7.X.0 GA](https://github.com/zigpy/zigpy/discussions/595#discussioncomment-277760),  [EZSP 6.8.X.0 GA](https://github.com/zigpy/zigpy/discussions/595#discussioncomment-282848),  [EZSP 6.9.X.0 GA](https://github.com/zigpy/zigpy/discussions/595#discussioncomment-282711),  [EZSP 6.10.X.0 GA](https://github.com/zigpy/zigpy/discussions/595#discussioncomment-890877)

**Comment by MattWestb** [comment]
### EZSP 6.7.0.0 - 6.7.10.0
Latest release [Zigbee EmberZNet SDK 6.7.10.0 GA](https://www.silabs.com/documents/public/release-notes/emberznet-release-notes-6.7.10.0.pdf) released with some bug fixes. 

#### Magure feature:

- Stack Protection

#### API news:

- MAC Timestamps
- emberIncomingNetworkStatusHandler
- emberOverrideAppendSourceRouteHandler
- emberOverrideIncomingRouteRecordHandler
- emberGetNeighborFrameCounter

GP:
- emberAfGreenPowerClusterCommissioningMessageStatusNotificationCallback
- emberAfGreenPowerClusterGetServerCommissioningSate
- emberAfPluginGreenPowerServerSinkTableAccessNotificationCallback
- emberAfGreenPowerClusterGpSinkCommissioningUnicastMode
- emberAfGreenPowerClusterGpSinkGpPairingUnicastMode
- emberAfGreenPowerClusterGpSinkCliCommissioningWindowExtend
- emberGpdAfPluginGetGpdIeeeCallback
- emberAfNcpGpSentHandler

#### API Changes:

- The emberAfExternalAttributeReadCallback() function's signature has changed
- Increased range of MacPollFailureWaitTime
- EZSP Protocol Version 8 and Secure EZSP Protocol Version 2
- Beacon Caching

#### Deprecated Items:
EFR32MG1 (1024 kb devices is not flagged).

Zigbee support – The upcoming Zigbee R23 specification will not be supported on any devices matching the following criteria:
• All EM35x devices
• Any devices with 256 kB flash or less [All wireless MCU families]

Maintenance Period: Critical bug fixes and security patches for the next major release implementing Zigbee specification R22 will continue to be made available for the lifetime of EM35xx series and wireless MCU devices with less than 256 kB, as specified by the wireless longevity commitment.

#### Removed Items
Removed in release 6.7.7.0

- The option to enable Serial 3 (USB CDC serial) for certain EM3xx parts has been removed from AppBuilder. This functionality has been deprecated and the backing code will be removed in a future release.

Removed in release 6.7.1.0

SDK no longer contains prebuilt application binaries for EM35x and 256k flash size devices except for the following:
• NCP images(ncp-spi, ncp-uart-hw, ncp-uart-sw) for
• EM3588
• efr32mg1p133f256gm48 - brd4150c
• efr32mg1p232f256gm48 - brd4151a
• Sniff image for EM3588
• xncp-commshub-uart-dual for efr32mg1p132f256gm48 – brd4155a
Applications may still be generated and built via Studio. 


**This is very likely being one golden release version of EZSP for EM3xx and EFR32MG1 and getting  Critical bug fixes and security patches until end of there Maintenance Period.**

Newer version of EZSP is implanting multi PAN and other things that is normally out of scope for our zigbee network so not interesting to using them.

#### Important bug fixes:

- 638989 Fixed an issue in the host-ncp sleepy end device configuration, where it continued short polling if the parent connectivity was lost temporarily during data transfer.

Comments: This bug is not fixed in newer versions of EZSP (6.8 and 6.9 ones) as of January 2021 !!
It is very likely braking pull control on many controller devices on Sonoff Zigbee Bridge and is making then draining there batteries.

Fixed in release 6.7.10.0:
Many fixes for association or rejoin process like _node being assigned an invalid address_ and _malformed link power delta message caused an assert_ in the Zigbee stack.

So this release is the (Re)joining fix release of EZSP 6.7.X.X.

Hope IKEA is using it in there new firmware for its controller that is in the pipe so the joining issues is being fixed (wish full thinking from my side).

PS: Silab have cooking EZSP 6.7.10.1 and 6.7.10.2 (if reading the internal posting in Silabs community) but is not public released so very likely only to some "gold customers" then its having more back ported fixes and some not public security fixes that im not allowed writing here for making it possible using EFR32MG1 devices that is not supported in EZSP 6.10.X.X.

**Comment by walthowd** [comment]
This would explain a few things `Fixed a corrupted message being detected as an address conflict in rare circumstances.`

**Reply by MattWestb** [reply]
The last can being one of the thing that is making IKEA controller draining batteries.
638989 `Fixed an issue in the host-ncp sleepy end device configuration, where it continued short polling if the parent connectivity was lost temporarily during data transfer.`

**Reply by MattWestb** [reply]
Stephan H have getting the info for his Sonoff Zigbee bridge so Hi can requesting updated firmware for it if hi like (and hopefully saving @Adminiuga hours of HA support cases).

**Reply by MattWestb** [reply]
Its also likely that the 638989 bug is more likely being triggered on second gen EFR32 chips then first gens (different CPU = different bugs in the the toochange / commpillers) but they have not writing that in the papirs (speculations from My side but still possible).

**Reply by MattWestb** [reply]
@walthowd You have getting one possibility testing if its working OK or not in real on EM357 hardware :-))
https://github.com/walthowd/husbzb-firmware/issues/12

**Comment by MattWestb** [comment]
### EZSP 6.9.2.0
Latest release [Zigbee EmberZNet SDK 6.9.2.0 GA](https://www.silabs.com/documents/public/release-notes/emberznet-release-notes-6.9.2.0.pdf) released with some bug fixes.
#### Magure feature:

- RTOS Common
- FreeRTOS

#### New API:

- Green Power Bidirectional Messaging
- emberAfPluginScenesServerCustomRecallSceneCallback()

#### Removed Items:
Removed in release 6.9.0.0

- All support for EM35x devices is removed. Additionally, Lightweight IP and Low-Voltage Shutdown plugins are removed.

##### RSSI Offset:

- The RSSI HWConf component can be used to configure the RSSI offset based on your part or board. On Series 1 and EFR32xG21 parts this offset is non-zero by default when enabled as those chips were shipped with a known offset. When enabling this consider carefully how this interacts with CCA checking and any CCA thresholds programmed into parts in the field via manufacturing tokens.

##### Zigbee Cluster Library

- The ZCL implementation is updated to align with ZCL8 with specific treatment for the following Zigbee Alliance CCBs:
1955 - Add Tunneling cluster Protocol IDs for IPv4 and IPv6
2310 - Error INVALID_VALUE for out-of-range Group ID
2427 - Default Response handling for Recall Scene command
2454 - BatteryAlarmState attribute is Reportable
2477 - ZCL status code cleanup
2543 - Complex attribute types (struct, array, set, bag) are UNREPORTABLE_ATTRIBUTE
2555 - Correct default for Window Covering cluster CurrentPositionLiftPercentage attribute
2560 - Correct default for Thermostat cluster Occupancy attribute
2722 - Basic ZCLVersion value is 8 beginning with ZCL8
2815 - Defaults values for Thermostat AC attributes
2823 - Temperature Measurement attribute defaults should be signed values
2893 - Treat Time cluster Time attribute as READ_ONLY if master bit in TimeStatus is set
2898 - Level cluster MoveToClosestFrequency command payload field type

Additionally, the global ClusterRevision attribute now reports the correct cluster revision for the attribute instance's respective cluster.
  
##### Door Lock Cluster

Support for the ZCL Door Lock cluster has been expanded to reflect capabilities described in the June 11 2020 Zigbee Alliance document 20-52431-003-Door Lock NFR changes to 14-0131-16-zcl-ch-7-closures-0611.docx. In summary:
• "Ajar" status in the Basic Information attribute and "Door ajar" in the AlarmMask attribute
• "Coerced User" in User type and "Coerced Alarm" in the AlarmMask attribute
• ZCL commands relating to Disposable User and Schedule
• ZCL commands relating to Biometric credentials
• Medium level for the SoundVolume attribute
 
##### ZCL Level Cluster
 
The SDK's definition of the ZIgbee Cluster Library Level cluster is expanded to include entries for the MinLevel, MaxLevel, CurrentFrequency, MinFrequency, and MaxFrequency attributes, and for the MoveToClosestFrequency command. 
These items are now presented in AppBuilder to assist developers who may wish to use them. However, the SDK's level-control plugin (sample code not intended for production use) has not been modified to support them.
 
**Comment:** 

Looks like all EFR32 first gen is still left in the studio but all recompiled interms is away for 256 kb or less device is not supported but can being compelled. 
The EFR32MG1x632/x732 is flagged Deprecated in 6.7.8.0 (1024kb status ??).

**Comment by MattWestb** [comment]
### EZSP 6.8.2.0
Latest release [Zigbee EmberZNet SDK 6.8.2.0 GA](https://www.silabs.com/documents/public/release-notes/emberznet-release-notes-6.8.2.0.pdf) released with some bug fixes. 

#### Magure feature:

- multi-PAN

#### API news:

- Stack Profile and Security Level
- slx_zigbee_routing_set_route_record_policy()

#### API Changes:

- Function signature for emberOverrideAppendSourceRouteHandler() has been changed
- The function signature, implementation, and invocations of emberAfAttributeValueSize() are modified to include an additional
- buffer size argument. 
- Buffer System

#### Deprecated Items:
Advanced Notice
Zigbee support – The upcoming Zigbee R23 specification will not be supported on any devices matching the following criteria:

- All EM35x devices
- Any devices with 256 kB flash or less [All wireless MCU families]

#### Removed Items:

- MSD bootloader binaries are no longer added to the GSDK package.
- Multi-network configuration is no longer supported on any 256k flash size devices (was previously available for EM357 and EFR32MG14).
- The EZSP command EZSP_OVERRIDE_CURRENT_CHANNEL has been removed. This command was a no-op on all architectures that support multi network functionality.

**Reply by Adminiuga** [reply]
there's 6.8.2.0 in this train

**Reply by MattWestb** [reply]
Sorry was not googling good enough :-((

Update i coming !!
Ed: Fixed :-)

**Comment by MattWestb** [comment]
### Found ZCL Rev 8 !!
I have not finding it on one "open source" and i normally dont need it but its start coming references to it so is some "finding it" somewhere please share it with our devs.

Latest "fund" version is [ZCL ver 7](https://github.com/Koenkk/zigbee-herdsman/blob/master/docs/Zigbee%20Cluster%20Library%20Specification%20v7.pdf)

Edit: Its now found, see comment below.

**Reply by MattWestb** [reply]
Latest fund of:

[Base Device Behavior Specification
Version 1.0](https://zigbeealliance.org/wp-content/uploads/2019/12/docs-13-0402-13-00zi-Base-Device-Behavior-Specification-2-1.pdf).

[ZigBee Lighting & Occupancy Device Specification Version 1.0](https://zigbeealliance.org/wp-content/uploads/2019/11/docs-15-0014-05-0plo-Lighting-OccupancyDevice-Specification-V1.0.pdf).

The latest is one "adopting" of ZLL to ZB3 and is one enchantment of ZCL so must reading both for getting all up to date.

**Reply by Hedda** [reply]
Did you manage to find Zigbee Cluster Library Specification Revision 8 (a.k.a. ZCL8 / ZCL v8 / ZCL r8 / ZCL 8)?

Curious if they added any additional substances to the Concentration Measurement clusters? ->  https://github.com/zigpy/zigpy/discussions/691

**Reply by MattWestb** [reply]
Nop I like getting it for Adminiuga and i have promising puddly that hi is getting more work implanting it then its being "found".

**Reply by MattWestb** [reply]
I have finding not the Zigbee alliance paper but extracted meta data for the zcl-8.0-07-5123-08.

Silabs have putting it together for chip for the ZAP (ZCL Advanced Platform) and in the meta data its written:
```
<configurator>
  <domain name="General" spec="zcl-8.0-07-5123-08" certifiable="true">
    <older spec="zcl-7.0-07-5123-07" certifiable="true"/>
    <older spec="zcl6-errata-14-0129-15"/>
    <older spec="zcl-6.0-15-02017-001"/>
    <older spec="zcl-1.0-07-5123-03"/>
  </domain>
```
Its one  part of ZAP: https://github.com/project-chip/zap/tree/master/zcl-builtin/silabs

All ZCL for Zigbeee and thred profiles and devices is in the matrix only need reading all the data in the right way.

**Reply by MattWestb** [reply]
## EXLUSIVE FOR @puddly ZCL R8
[Cluster Library Specification
Document 07-5123 Revision 8
Chapter Document 14-0125](https://zigbeealliance.org/wp-content/uploads/2021/10/07-5123-08-Zigbee-Cluster-Library.pdf)

Also as uploaded if being deleted on the CSA site.
[07-5123-08-Zigbee-Cluster-Library.pdf](https://github.com/zigpy/zigpy/files/7553476/07-5123-08-Zigbee-Cluster-Library.pdf)

**Reply by puddly** [reply]
Thank you!

This release doesn't include any major changes from v7 as far as I can tell, other than the addition of a few new clusters. I'll add those to my ~~v7~~ v8 PR.

**Reply by MattWestb** [reply]
I was looking on OnOff, Light level and Thermostat and no large things only that all cluster shall have version added so can see if its one old or ZCL 7 / 8 that the cluster is defined.

Also some error response is deleted or flagged for deletion.

Next up to getting for Alexei is Zigbee R23 so hi can cooking little more with EZSP :-))

**Reply by MattWestb** [reply]
Sorry for being picky its not V8 its R8  (that was one of the problem to finding it) ;-)

**Comment by MattWestb** [comment]
## Silabs hardware reference collections
### EM35x
[EM35X KBs:](https://www.silabs.com/community/wireless/zigbee-and-thread/knowledge-base.entry.html/2020/04/02/_20_em35x-0Y2P)
[Telegesis Legacy Module Firmware](https://www.silabs.com/community/wireless/zigbee-and-thread/knowledge-base.entry.html/2020/04/27/telegesis_legacymodulefirmware-3TNm)

## EFR32 Series 1
[Master Wireless Hardware KBA list for EFR32 Series 1 SoCs and Wireless Modules](https://www.silabs.com/community/wireless/zigbee-and-thread/knowledge-base.entry.html/2020/10/30/master_wireless_hardwarekbalistforefr32series-Z65t)

## EFR32 Series 2
[Master Wireless Hardware Doc list for EFR32 Series 2 SoCs and Wireless Modules](https://www.silabs.com/community/wireless/zigbee-and-thread/knowledge-base.entry.html/2020/11/19/master_wireless_hardwarekbalistforefr32series-c2wI)

**Comment by MattWestb** [comment]
# CEL MeshConnect™ EM35x USB Sticks
Was finding the programming manual that is pointing out what GPIO is used and for what.
[3388832429388690em357_usb_sticks_ds.pdf](https://github.com/zigpy/zigpy/files/6409608/3388832429388690em357_usb_sticks_ds.pdf)

**Comment by MattWestb** [comment]
## ZIGBEE HOME AUTOMATION PUBLIC APPLICATION PROFILE
Home Automation
Public Application Profile
ZigBee Profile: 0x0104
Revision 29
Version 1.2

[075367r03ZB_AFG-Home_Automation_Profile_for_Public_Download.pdf](https://github.com/zigpy/zigpy/files/6508541/075367r03ZB_AFG-Home_Automation_Profile_for_Public_Download.pdf)

**Comment by MattWestb** [comment]
# de(F)CONZ Serial protocol

[deCONZ-Serial-Protocol-en_1.20.pdf](https://deconz.dresden-elektronik.de/raspbian/deCONZ-Serial-Protocol-en_1.20.pdf?ref=gh)

**Reply by MattWestb** [reply]
Upcoming firmware change: Enable APS ACKs for ZCL Default Response.
Moving ACK to the firmware then deCONZ is not fast enough for doing "on the fly" but not for OTA related frames(?!?).
https://github.com/dresden-elektronik/deconz-rest-plugin/pull/5173#issuecomment-894281514

Its looks being one good move then i have sniffing my RaspBee I it was very lazy with ACK and devices was re sending frames many times then the host system was not acking the frames.

Impact on Zigpy-deconz ??

PS: Before deCONZ have nearly not sending any default response at all also for ZCL frames that haven making much problems also with direct connected SED and many more devices so they is learning somthing in Dresden :-))

**Reply by MattWestb** [reply]
Ops updated  docks [deCONZ-Serial-Protocol-en_1.21.pdf](https://deconz.dresden-elektronik.de/raspbian/deCONZ-Serial-Protocol-en_1.21.pdf)

And the [Firmware Changelog](https://github.com/dresden-elektronik/deconz-rest-plugin/wiki/Firmware-Changelog) for deCONZ radios.

**Reply by MattWestb** [reply]
Rasp/CornBee I:
ZCL Default Responses for ZCL in the NCP and Support custom Node Descriptor manufacturer code.

Rasp/CornBee II:

The same as first gen and Improved stability of the UART communication, Fix loopbug behaviour caused by a USB stack bug and Improved serial communication performance.

The USB problems is hopefully fixed that have generating many issues in ZHA and the lazy default response is also wanted and hope the NCP is being faster sending IEEE 802.15.4 Ack packages that my RaspBee I is very lazy making that and can making problems with network stability and problem with sleeping end device (Philips HUE and IKEA remotes draining there batteries if being direct children).

They still have some very old bug left like that must restart the system after some uptime for joining new devices that is still one PIA.

And the second gen is sill missing touch link in the firmware (firs gen have and working great).

**Comment by MattWestb** [comment]
### EZSP 6.10.0.0
Latest release [Zigbee EmberZNet SDK 6.10.2.0 GA](https://www.silabs.com/documents/public/release-notes/emberznet-release-notes-6.10.2.0.pdf). 

#### Magure feature:

- Support for redesigned MG22 radio board (BRD4183C)
- Updated IAR and GCC compiler versions

#### API news:

- emberAddChild() - SoC-only API for adding a child to the child/neighbor table, allowing direct manipulation of these tables by the application.
- emberRemoveChild() - SoC-only API for removing a node from child/neighbor table, allowing direct manipulation of these tables by the application.
- emberRemoveNeighbor() - SoC-only API for removing a node from neighbor table, allowing direct manipulation of these tables by the application.

#### API Changes:

Not of interest.

#### Deprecated Items:

Not of interest.

#### Removed Items:

Not of interest.

### More interesting is fixed issues in Gecko Platform 3.2.0.0 GA.

- 703231 Fixed an issue in Power Manager on Series 2 devices where we could get stuck waiting for HFXO to be ready when a hardware request on HFXO was removed in the middle of the restore process in Power Manager.
- 675253 Fixed an issue in Power Manager where, if the HFXO startup failed while actively waiting for it to be ready in a critical section, we could wait indefinitely
- 697483 Fixed bug affecting CTUNE for HFXO on Series 2 devices.

Hopefully no new bugs in the new IAR and GCC compiler that is the normal then stepping them up.

Latest found [Gecko Platform 3.2.2.0 GA](https://www.silabs.com/documents/public/release-notes/gecko-platform-release-notes-3.2.2.0.pdf).

**Reply by MattWestb** [reply]
@tube0013 The 2 first fixes i think is not relevant but the last can being one problems also for your NCPs.

I have updating Si5 and its was not updating the tool change but was saying all was up to date but the compiling it was using the old one and was showing the new as preferred and used then making new projects so taking one extra lock in installed components for the updated GCC so its being installed and used for six ten projects !!

New bootloader :-))
```
Gecko Bootloader v1.12.00
1. upload gbl
2. run
3. ebl info
BL > 
```
And also MG22 Zigbee Control Bridge :-))))
```
stack ver. [6.10.0 GA build 169]
```

Next six ten NCP for Billy and the tuya / LIDL MG21 module !!

**Reply by tube0013** [reply]
Thanks, I'll take a look in a few days currently away for a long weekend.

**Reply by MattWestb** [reply]
Zigbee EmberZNet SDK [6.10.1.0 GA](https://www.silabs.com/documents/public/release-notes/emberznet-release-notes-6.10.1.0.pdf) First bug fix released but i have not finding any interesting in it.

Gecko Platform 3.2. have getting patches 2 but the release notes is not made public only accessible thru SS5 so not posting here.

**Comment by MattWestb** [comment]
## Zigbee EmberZNet SDK 7.0.2.0 GA Gecko SDK Suite 4.0
Only 2 interesting issues found in the docks:

- 363162
  - There is a bug in emberAfAddAddressTableEntry
which could allow for duplicate entries in the
address table
  - Under Investigation

- 631713
   - A Zigbee End Device will report address conflicts
repeatedly if the plugin "Zigbee PRO Stack Library"
is used instead of "Zigbee PRO Leaf Library".
    - Use the"Zigbee PRO Leaf Library" instead of the "Zigbee PRO
Stack Library" plugin.

I dont knowing how "old" this bugs is if its was there for the time it was started or implanted on the road.

The first i think can being in both devices and in NCPs and can making that many devices is reporting address conflicts.
The second shall not being seen if the work around is applyed and SED shall using the Leaf Library and not sleeper the normal but the problem is that no sleepers like no neutral switches can using both and is using the wrong one and getting this problem and making problems in the network.

https://www.silabs.com/documents/public/release-notes/emberznet-release-notes-6.10.2.0.pdf

**Comment by MattWestb** [comment]
Zigbee EmberZNet SDK 7.2.0.0 GA and Gecko SDK Suite 4.2 is released December 14, 2022 but not posted online only in SS5 install local files.

In EZSP 7.2.0.0 is many interesting bug fixes but i dont knowing if all is relevant for our system.
I think i can being interesting for @puddly reading fixes and known issues then its may fixes for direct children (re)joining things.
Also i think @Adminiuga is knowing more if some of the fixes helping fixing some of the problem we have seen in ZHA with (re)joining, routing and network stability if users i updating there coordinator to EZSP 7.2.0.0.

**Reply by tube0013** [reply]
Have you seen emberznet release notes? I get a 404 when I try to get to them.

**Reply by MattWestb** [reply]
I know all links is broken but SS5 is installing them in your SS5 directory `/offline/com.silabs.sdk.stack.super_4.2.0/protocol/zigbee/documentation/[zigbee-release-notes.pdf]`
You only need updating SS5 for getting them ;-))

**Reply by Hedda** [reply]
FYI, release notes links are now available for Silicon Labs Gecko SDK 4.2.0 + Zigbee EmberZNet SDK 7.2.0.0 GA:

https://github.com/SiliconLabs/gecko_sdk/releases/tag/v4.2.0

Zigbee EmberZNet 7.2 looks to also use EZSP v9 (EmberZNet Serial Protocol version 9) same as Zigbee EmberZNet 7.1 or?

https://www.silabs.com/documents/public/release-notes/emberznet-release-notes-7.2.0.0.pdf

New features seams to be mostly related to Dynamic Multiprotocol and multi-PAN 802.15.4 in RCP mode:

- Secure key storage support for MG2x parts that support Secure Vault-High
- Dynamic Multiprotocol Bluetooth and multi-PAN 802.15.4 in RCP mode
- Dynamic Multiprotocol Zigbee-NCP + Bluetooth-NCP (experimental)
- Manufacturing Library (MfgLib) support for Concurrent Multiprotocol RCP
- Zigbee + OpenThread Concurrent Lis-tening on MG24 parts (experimental)
- Zigbee on Host (ZigbeeD) support for 32 bit and 64 bit x86 architecture (experimental)
- MGM240S SiP Module Support

They also added simultaneous Dual-Band and Dual-PHY for Zigbee Smart Energy on MG24/MG12:

- MG24+Si4468 Dual-PHY Zigbee Smart Energy support
- MG12 Dual-Band 2.4GHz + SubGHz Zigbee Smart Energy support
PS: Silicon Labs also released Gecko SDK (GSDK) Version 4.1.4.0 as a bug-fix release that contains Zigbee EmberZNet SDK 7.1.4.0:

Deprecated in release 7.2.0.0:

- The Secure EZSP feature will be removed in a future release.

PS: Silicon Labs also released Gecko SDK (GSDK) Version 4.1.4.0 as a bug-fix release that contains Zigbee EmberZNet SDK 7.1.4.0:

https://github.com/SiliconLabs/gecko_sdk/releases/tag/v4.1.4

https://www.silabs.com/documents/public/release-notes/emberznet-release-notes-7.1.4.0.pdf

**Comment by MattWestb** [comment]
## Zigbee R23 for @Adminiuga  !!
and our @puddly :-))
Its one temp like and if is not working do one new download request for getting one working one.

The nickname is **Zigbee PRO 2023**.


https://csa-iot.org/wp-content/uploads/2023/04/05-3474-23-csg-zigbee-specification-compressed.pdf

**Comment by Hedda** [comment]
ZBOSS NCP Serial Protocol (v1.5) prepared by DSR Corporation for ZOI:

https://cloud.dsr-corporation.com/index.php/s/BAn4LtRWbJjFiAm

(link posted by [eexarevsky](https://github.com/eexarevsky) in https://github.com/zigpy/zigpy/issues/394)
